Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,411 @@
+                                            Name = o.Name,
+                                            Logo = o.Logo
+                                        }).FirstOrDefault(),
+                                    TeamScore = matchTeamJoin.Score,
+                                    OpposingTeamScore = matchTeamJoin.OpposingTeamScore,
+                                    Status = matchTeamJoin.Status,
+                                    MatchStatusID = match.StatusID,
+                                    Time = match.Time
+                                }).AsEnumerable();
+            return vm;
+        }
+        public IEnumerable<MatchExtendedViewModel> GetMatchBySeriesID(int seriesID)
+        {
+            var match = _context.Match
+                        .Where(m => m.SeriesID == seriesID)
+                        .Select(m => new MatchExtendedViewModel
+                        {
+                            ID = m.ID,
+                            Time = m.Time,
+                            Surface = _context.Surface
+                                        .FirstOrDefault(s => s.ID == m.SurfaceID)
+                                        .Description,
+                            Status = _context.MatchStatus
+                                        .FirstOrDefault(ms => ms.ID == m.StatusID)
+                                        .Description,
+                            Team = m.MatchTeam.Select(mt => new TeamExtendedViewModel
+                            {
+                                ID = mt.TeamID,
+                                Name = _context.Team
+                                   .FirstOrDefault(t => t.ID == mt.TeamID)
+                                   .Name,
+                                TeamRecord = _context.SeriesTeam
+                                           .Where(st => st.SeriesID == m.SeriesID && st.TeamID == mt.TeamID)
+                                           .Select(st => new TeamRecordViewModel
+                                           {
+                                               Win = st.Win,
+                                               Loss = st.Loss,
+                                           })
+                            })
+                        });
+            return match;
+        }
+        public async Task<MatchViewModel> PostMatchAsync(MatchUploadViewModel m)
+        {
+            var match = new Match
+            {
+                SeriesID = m.SeriesID,
+                Time = m.StartTime,
+                SurfaceID = m.SurfaceID ?? 1, //unknown surface id
+                StatusID = 1 //default unconfirmed
+            };
+            await _context.Match.AddAsync(match);
+            await _context.SaveChangesAsync();
+
+            foreach (var teamID in m.TeamIDs)
+            {
+                await CreateMatchTeamMappingAsync(teamID, match.ID);
+            }
+            await _context.SaveChangesAsync();
```

Comment:
You should only need to call `SaveChanges` once here. That way, this whole operation becomes one transaction and it either all succeeds or none of it succeeds.